core/common/src/main/java/alluxio/security/LoginUser.java

```diff
@@ -50,18 +50,21 @@ private LoginUser() {} // prevent instantiation
    * @return the login user
    * @throws IOException if login fails
    */
-  public static User get() throws IOException {
+  public static synchronized User get() throws IOException {
```

Comment:
Would making entire method `synchronized` cause a bit worse performance? in most cases, `sLoginUser` is not null and would not need to grab the lock.